components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -518,7 +518,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ResponseError(ErrorCodes.PERMISSION_DENIED, "Cannot (re-)start a deleted workspace.");
         }
         const userEnvVars = this.userDB.getEnvVars(user.id);
-        let projectEnvVarsPromise = this.internalGetPublicProjectEnvVars(workspace.projectId);
+        let projectEnvVarsPromise = this.internalGetProjectEnvVars(workspace.projectId);
```

Comment:
With the filtering moved to another level, I believe you can delete the `internalGet[...]ProjectEnvVars` helper function, and just call this here:

```suggestion
        let projectEnvVarsPromise = this.projectsService.getProjectEnvironmentVariables(workspace.projectId);
```